engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/details/views/repository-details.html

```diff
@@ -32,6 +32,22 @@ <h2 translate>{{ repository.name }}</h2>
           </span>
         </li>
 
+        <li role="menuitem">
+          <a ng-click="republishRepository(repository)"
+             ng-hide="syncInProgress(repository.last_sync) || denied('edit_products', product)"
+             translate>
+            Republish Repository Metadata
+          </a>
+
+          <span class="disabled" ng-show="syncInProgress(repository.last_sync)" translate>
+            Cannot republish Repository, a sync is already in progress.
+          </span>
+
+          <span class="disabled" ng-show="denied('sync_products', product)" translate>
+            Cannot republish Repository without the proper permissions.
+          </span>
+        </li>
```

Comment:
Why do we need all three of these states? The second state would seem to make sense as a tooltip rather than as a full blown menu item. I think in some cases I'd rather hit the button if I have permission and then be told a sync is already in progress with a link to the currently running sync so I don't have to go hunt it down to track it.